apps/server/src/shared/infra/oauth-provider/hydra/hydra.service.ts

```diff
@@ -18,12 +27,36 @@ export class HydraService extends OauthProviderService {
 		this.hydraUri = Configuration.get('HYDRA_URI') as string;
 	}
 
-	async acceptLogoutRequest(challenge: string): Promise<RedirectResponse> {
+	acceptConsentRequest(challenge: string, body: AcceptConsentRequestBody): Promise<ProviderRedirectResponse> {
+		return this.put<ProviderRedirectResponse>('consent', 'accept', challenge, body);
+	}
+
+	acceptLoginRequest(challenge: string, body: AcceptLoginRequestBody): Promise<ProviderRedirectResponse> {
+		return this.request<ProviderRedirectResponse>(
+			'PUT',
+			`${this.hydraUri}/oauth2/auth/requests/login/accept?login_challenge=${challenge}`,
+			body
+		);
+	}
+
+	async acceptLogoutRequest(challenge: string): Promise<ProviderRedirectResponse> {
 		const url = `${this.hydraUri}/oauth2/auth/requests/logout/accept?logout_challenge=${challenge}`;
-		const response: Promise<RedirectResponse> = this.request<RedirectResponse>('PUT', url);
+		const response: Promise<ProviderRedirectResponse> = this.request<ProviderRedirectResponse>('PUT', url);
+		return response;
+	}
+
+	getConsentRequest(challenge: string): Promise<ProviderConsentResponse> {
+		const response: Promise<ProviderConsentResponse> = this.get<ProviderConsentResponse>('consent', challenge);
 		return response;
 	}
 
+	getLoginRequest(challenge: string): Promise<ProviderLoginResponse> {
+		return this.request<ProviderLoginResponse>(
+			'GET',
+			`${this.hydraUri}/oauth2/auth/requests/login?login_challenge=${challenge}`
```

Comment:
refactor to use generalized method put<T>

Reply:
done